src/celeritas/optical/CoreTrackView.hh

```diff
@@ -88,6 +96,42 @@ CoreTrackView::CoreTrackView(ParamsRef const& params,
     CELER_EXPECT(track_slot_id_ < states_.size());
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Initialize the track states.
+ */
+CELER_FUNCTION CoreTrackView&
+CoreTrackView::operator=(TrackInitializer const& init)
+{
+    // Initialiize the sim state
+    this->sim() = SimTrackView::Initializer{init.time};
+
+    // Initialize the geometry state
+    auto geo = this->geometry();
+    geo = GeoTrackInitializer{init.position, init.direction};
+    if (CELER_UNLIKELY(geo.failed() || geo.is_outside()))
+    {
+#if !CELER_DEVICE_COMPILE
+        if (geo.is_outside())
+        {
+            // Print an error message if initialization was "successful" but
+            // track is outside
+            CELER_LOG_LOCAL(error) << "Track started outside the geometry";
+        }
+#endif
+        this->apply_errored();
+        return *this;
+    }
+
+    // Initialize the particle state
+    this->particle()
+        = ParticleTrackView::Initializer{init.energy, init.polarization};
+
+    //! \todo Add physics view and clear physics state
+
+    return *this;
+}
```

Comment:
I like it! Future work for the main core track view?  (And would we need a second version for the "detailed initializer" that has the full volume state?)

Reply:
Yep! And yeah, gotta think about the best way to handle the special geo initialization...